mscore/importgtp-gp6.cpp

```diff
@@ -521,41 +522,32 @@ void GuitarPro6::readTracks(QDomNode* track)
                               s->setBarLineSpan(2);
                               }
                         }
-                  else if (nodeName == "Properties") {
-                        QDomNode currentProperty = currentNode.firstChild();
-                        while (!currentProperty.isNull()) {
-                              QString propertyName = currentProperty.attributes().namedItem("name").toAttr().value();
-                              if (!propertyName.compare("Tuning")) {
-                                    // set up the tuning for the part
-                                    QString tuningString         = currentProperty.firstChild().toElement().text();
-                                    QStringList tuningStringList = tuningString.split(" ");
-                                    int strings                  = 0;
-                                    std::vector<int> tuning(tuningStringList.length());
-                                    //int tuning[tuningStringList.length()];
-                                    int frets = 24;
-                                    for (auto iter = tuningStringList.begin(); iter != tuningStringList.end(); ++iter) {
-                                          int currentString = (*iter).toInt();
-                                          tuning[strings] = currentString;
-                                          strings++;
-                                          }
-                                    StringData* stringData = new StringData(frets, strings, &tuning[0]);
-                                    Instrument* instr      = part->instrument();
-                                    instr->setStringData(*stringData);
-                                    instr->setSingleNoteDynamics(false);
-                                    hasTuning = true;
-                                    createTuningString(strings, &tuning[0]);
-                                    }
-                              else if (!propertyName.compare("DiagramCollection")) {
-                                    QDomNode items       = currentProperty.firstChild();
-                                    QDomNode currentItem = items.firstChild();
-                                    while (!currentItem.isNull()) {
-                                          readChord(&currentItem, trackCounter);
-                                          currentItem = currentItem.nextSibling();
-                                          }
-                                    }
-                              currentProperty = currentProperty.nextSibling();
+                  // GP6 only
+                  else if (nodeName == "PartSounding")
+                        part->instrument()->setTranspose(Interval(currentNode.firstChildElement("TranspositionPitch").text().toInt()));
+                  // GP7 only
+                  else if (nodeName == "Transpose")
```

Comment:
The same applies to this code: it is better not to put GP7-related code to a class named `GuitarPro6`, and I guess it would be more difficult to extract this code from here in the future.